src/csharp/Microsoft.Spark.Worker/Processor/PayloadProcessor.cs

```diff
@@ -56,6 +56,7 @@ internal Payload Process(Stream stream)
             payload.Version = SerDe.ReadString(stream);
             payload.TaskContext = new TaskContextProcessor(_version).Process(stream);
             payload.SparkFilesDir = SerDe.ReadString(stream);
+            SparkFiles.SetRootDirectory(payload.SparkFilesDir);
```

Comment:
Does this even work in cluster mode? There is no listener on the worker nodes. You should just leverage the Python env variables used in registering the UDF.

Reply:
This was how python did it.  This should set the private variables on the worker.   When the worker calls SparkFIles.getRootDirectory, it will won't call the DotnetBackend to fulfill the request, but use the private variables that were set.

Reply:
Ah OK. This should be fine, but not sure which approach is cleaner. Now, you have to do more stuff to set up the tests, etc.

Reply:
I've tested on local mode.  I'll still have to test this out on cluster mode which may take some time.

Reply:
Oh I meant changes in unit tests.

Reply:
Meant to address your earlier comment regarding if this was tested in Cluster Mode.